openstack/tools/openstack_regression_tests_runner.sh

```diff
@@ -0,0 +1,143 @@
+#!/bin/bash -eu
+#
+# Run Openstack regression tests.
+#
+FUNC_TEST_PR=
+FUNC_TEST_TARGET=
+IMAGES_PATH=$HOME/tmp
+
+usage () {
+    cat << EOF
+USAGE: `basename $0` OPTIONS
+
+Run Openstack regression tests.
+
+OPTIONS:
+    --func-test-target TARGET_NAME
+        Provide the name of a specific test target to run.
+    --func-test-pr PR_ID
+        Provides similar functionality to Func-Test-Pr in commit message. Set
+        to zaza-openstack-tests Pull Request ID.
+    --help
+        This help message.
+EOF
+}
+
+while (($# > 0)); do
+    case "$1" in
+        --debug)
+            set -x
+            ;;
+        --func-test-target)
+            FUNC_TEST_TARGET=$2
+            shift
+            ;;
+        --func-test-pr)
+            FUNC_TEST_PR=$2
+            shift
+            ;;
+        --help|-h)
+            usage
+            exit 0
+            ;;
+        *)
+            echo "ERROR: invalid input '$1'"
+            usage
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+if [[ -z $FUNC_TEST_TARGET ]]; then
+   echo "ERROR: must provide a target name with --func-test-target"
+   exit 1
+fi
+
+# This is required for magnum tests and zaza will look in swift if it is not cached so we need to cache it first.
+mkdir -p $IMAGES_PATH
+if ! [[ -e $IMAGES_PATH/fedora-coreos-35.qcow2 ]]; then
```

Comment:
`-f` is better here. Also can move the "not" inside:
```
if [[ ! -f $IMAGES_PATH/fedora-coreos-35.qcow2 ]]
```

Reply:
done.